File: Sources/BeagleUI/Sources/DeclarativeViews/UIComponents/Image/NetworkImage.swift
//
//  NetworkImage.swift
//  BeagleUI
//
//  Created by Eduardo Sanches Bocato on 14/10/19.
//  Copyright © 2019 Daniel Tes. All rights reserved.
//

import Foundation

public struct NetworkImage: Widget {
    
    public let url: String
    public let contentMode: ImageContentMode
    
    // MARK: - Initialization
    
    public init(
        url: String,
        contentMode: ImageContentMode = .fitCenter
    ) {
        self.url = url
        self.contentMode = contentMode
    }
    
}

File: beagle-ui/src/main/java/br/com/zup/beagleui/framework/widget/ui/NetworkImage.kt
package br.com.zup.beagleui.framework.widget.ui

import br.com.zup.beagleui.framework.widget.core.ImageContentMode
import br.com.zup.beagleui.framework.widget.core.Widget

data class NetworkImage(
    val url: String,
    val contentMode: ImageContentMode = ImageContentMode.FIT_CENTER
) : Widget
